client/app/bundles/course/video/submission/containers/DiscussionElements/Editor.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReactSummernote from 'react-summernote';
+import RaisedButton from 'material-ui/RaisedButton';
+
+import style from '../Discussion.scss';
+
+const editorOptions = {
+  airMode: true,
+  dialogsInBody: false,
+  popover: {
+    air: [
+      ['style', ['style']],
+      ['font', ['bold', 'underline', 'clear']],
+      ['script', ['superscript', 'subscript']],
+      ['color', ['color']],
+      ['para', ['ul', 'ol', 'paragraph']],
+      ['table', ['table']],
+      ['insert', ['link', 'picture']],
+    ],
+  },
+};
+
+const propTypes = {
+  content: PropTypes.string.isRequired,
+  disabled: PropTypes.bool,
+  showCancel: PropTypes.bool,
+  cancelButtonText: PropTypes.string,
+  submitButtonText: PropTypes.string,
+  children: React.PropTypes.element,
+  onSubmit: PropTypes.func,
+  onCancel: PropTypes.func,
+  onContentUpdate: PropTypes.func,
+};
+
+const defaultProps = {
+  disabled: false,
+  showCancel: true,
+  cancelButtonText: 'Cancel',
```

Comment:
The strings in the frontend need to be translated. We use ReactIntl for translations.